app/models/user.rb

```diff
@@ -235,12 +235,15 @@ def self.try_to_login(login, password, api_request = false)
       if api_request && user.authenticate_by_personal_access_token(password)
         logger.debug("Authenticated user #{user.login} with a Personal Access Token.")
 
-        unless user.auth_source.valid_user?(user.login)
+        if user.auth_source.valid_user?(user.login)
+          user.post_successful_login
```

Comment:
can't we push this to the end of the method and do it only once?
The branch where we check if user isn't nil before final return would be perfect place I belive :)

Reply:
Good point :+1: